src/NewsFeed/NewsFeed.tsx

```diff
@@ -36,16 +40,14 @@ function NewsFeed() {
       .then((response) => response.json())
       .then(
         (data) => {
-          const randomStoryIds: string[] = getRandom(data, 10); // Assuming getRandom returns an array of story IDs as strings
+          const randomStoryIds: string[] = getRandom(data, 10); // get an array of 10 story IDs as strings
 
-          // Construct URLs from story IDs
-          storyUrls = randomStoryIds.map((storyId: string) => {
+          // Construct URLs with the randomStoryIds
+          const storyUrls: string[] = randomStoryIds.map((storyId: string) => {
             return `https://hacker-news.firebaseio.com/v0/item/${storyId}.json`;
           });
 
-          setLoading(1);
-          // setRandomStories(randomStoryIds);
-          setRandomStoriesData(storyUrls);
+          setLoading(1); // let user know something is happening
           randomStoriesData = storyUrls;
```

Comment:
Should be using `setRandomStoriesData`, if you don't get the data in the current rendering them problem might be [here](https://react.dev/reference/react/useState#updating-state-based-on-the-previous-state>). Same is valid for `randomStoriesContent`